pipeline/test/end-to-end/test_scoring.jl

```diff
@@ -0,0 +1,104 @@
+using DrWatson
+quickactivate(@__DIR__(), "EpiAwarePipeline")
+
+@testset "run inference for random scenario and do scoring" begin
+    using EpiAwarePipeline, EpiAware, Plots, Statistics, RCall, DataFramesMeta
+    pipeline = EpiAwareExamplePipeline()
+    prior = RtwithoutRenewalPriorPipeline()
+
+    ## Set up data generation on a random scenario
+
+    missing_padding = 14
+    lookahead = 21
+    n_observation_steps = 35
+    tspan_gen = (1, n_observation_steps + lookahead + missing_padding)
+    tspan_inf = (1, n_observation_steps + missing_padding)
+    inference_method = make_inference_method(pipeline; ndraws = 2000)
+    truth_data_config = make_truth_data_configs(pipeline)[1]
+    inference_configs = make_inference_configs(pipeline)
+    inference_config = rand(inference_configs)
+
+    ## Generate truth data and plot
+    truth_sampling = generate_inference_results(inference_config, prior; tspan = tspan_gen)
+    truthdata = truth_sampling["inference_results"].generated.generated_y_t
+    ##
+    plt = scatter(truthdata, xlabel = "t", ylabel = "y_t", label = "truth data")
+    vline!(plt, [n_observation_steps + missing_padding + 0.5], label = "forecast start")
+
+    ##Generate true Rt values
+    truth_It = truth_sampling["inference_results"].generated.I_t
+    truth_Rt = expected_Rt(truth_sampling["epiprob"].epi_model.data,
+        truth_sampling["inference_results"].generated.I_t)
+    plt_Rt = plot(truth_Rt, xlabel = "t", ylabel = "R_t", label = "truth Rt")
+    vline!(plt_Rt, [n_observation_steps + missing_padding + 0.5], label = "forecast start")
+
+    ##
+    ## Run inference
+    obs_truthdata = truthdata[tspan_inf[1]:tspan_inf[2]]
+
+    inference_results = generate_inference_results(
+        Dict("y_t" => obs_truthdata, "truth_gi_mean" => inference_config["gi_mean"]),
+        inference_config, pipeline; tspan = tspan_inf, inference_method)
+
+    ## Make 21-day forecast
+
+    forecast_quantities = generate_forecasts(inference_results, lookahead)
+    forecast_y_t = mapreduce(hcat, forecast_quantities.generated) do gen
+        gen.generated_y_t
+    end
+    forecast_qs = mapreduce(hcat, [0.025, 0.25, 0.5, 0.75, 0.975]) do q
+        map(eachrow(forecast_y_t)) do row
+            if any(ismissing, row)
+                missing
+            else
+                quantile(row, q)
+            end
+        end
+    end
+    plt = scatter(truthdata, xlabel = "t", ylabel = "y_t", label = "truth data")
+    vline!(plt, [n_observation_steps + missing_padding + 0.5], label = "forecast start")
```

Comment:
why not reuse truth_gen here it seems like unnecessary room for error

Reply:
Good spot